[examples] Added `with-couchbase` example

ejscribner opened the pull request:
This adds a simple `with-couchbase` example app complete with a check for successful connection (in `index.js`).

**Note:** To check the connection, this application runs a simple KV GET operation with a key that does not exist and expects a response of `document not found`. Any other response indicates a failure to connect.

Thank you!

ejscribner commented:
@leerob Any chance we could get a re-review on this? Updated a few things and simplified the `index.js` page to make it easier for first time users.

leerob commented on examples/with-couchbase/package.json:
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "with-couchbase",
+  "version": "0.1.0",
```
```suggestion
  "private": true,
```

leerob commented on examples/with-couchbase/pages/index.js:
```diff
@@ -0,0 +1,254 @@
+        </a>
+      </footer>
+
+      <style jsx>{`
```
Can this use CSS Modules instead of styled-jsx like what create-next-app ships with? Thanks!

leerob commented on examples/with-couchbase/util/couchbase.js:
```diff
@@ -0,0 +1,65 @@
+const couchbase = require('couchbase')
```
Can this use ESM?

ejscribner commented:
Updated to use ESM to import couchbase, added `"private": true, ` to `package.json`, and upgraded to use CSS modules. Thanks for the feedback @leerob !!

leerob commented on examples/with-couchbase/package.json:
```diff
@@ -1,6 +1,7 @@
 {
   "name": "with-couchbase",
   "version": "0.1.0",
```
```suggestion
```

These need to be remove to follow the examples format

leerob commented on examples/with-couchbase/package.json:
```diff
@@ -0,0 +1,17 @@
+{
+  "name": "with-couchbase",
+  "version": "0.1.0",
```
```suggestion
```

These need to be remove to follow the examples format

leerob commented on examples/with-couchbase/README.md:
```diff
@@ -0,0 +1,95 @@
+## Example app using Couchbase
+
+[Couchbase](https://www.couchbase.com/) is The Modern Database for Enterprise Applications. This example will show you how to connect to and use Couchbase for your Next.js app.
```
```suggestion
[Couchbase](https://www.couchbase.com/) is a modern database for enterprise applications. This example will show you how to connect to and use Couchbase in your Next.js app.
```

leerob commented on examples/with-couchbase/README.md:
```diff
@@ -0,0 +1,95 @@
+
+Set each variable on `.env.local`:
+
+- `COUCHBASE_USERNAME` - The username of an authorized user on your Couchbase instance
```
Note: It'd be great if there was an integration that took care of this for you 😄 

https://vercel.com/integrations#databases

ejscribner replied:
Will add this ASAP!

leerob replied:
- https://vercel.com/docs/integrations
- https://github.com/vercel/example-integration

Let me know if you have any questions!

leerob commented on examples/with-couchbase/package.json:
```diff
@@ -0,0 +1,15 @@
+    "next": "latest",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2"
+  },
+  "license": "MIT"
+}
```
```suggestion
  }
}
```
No license either 👍
